x-pack/plugins/security_solution/common/endpoint/types/actions.ts

```diff
@@ -75,12 +90,13 @@ export interface LogsEndpointAction {
  * An Action response written by the endpoint to the Endpoint `.logs-endpoint.action.responses` datastream
  * @since v7.16
  */
-export interface LogsEndpointActionResponse {
+export interface LogsEndpointActionResponse<TOutputContent extends object = object> {
   '@timestamp': string;
   agent: {
     id: string | string[];
   };
-  EndpointActions: EndpointActionFields & ActionResponseFields;
+  EndpointActions: EndpointActionFields &
+    ActionResponseFields & { output?: ActionResponseOutput<TOutputContent> };
```

Comment:
So these types are supposed to mimic the index mappings we have on the endpoint index. So this addition doesn't look right. Why is it that we're adding an output under `EndpointActions`? We would also need to update the endpoint index mapping if we want this here.

As I understood, the `output` key should be part of `ActionDetails`.

Reply:
`output` is gonna be filled by the endpoint depending on the action response. I think we don't need to map everything under `EndpointActions`, @pzl or @joeypoon can answer this.

cc: @paul-tavares @kevinlog

Reply:
There is no technological reason why TS types must === mapping.

Unmapped fields are still stored, are still JSON, will still be sent to the frontend who can parse the JSON and therefore know about the property.

I don't know the rest of the context here, whether this type is being used in a filter somewhere that will otherwise inform part of the UI what filters are available (unmapped means not filterable), or if there are similar side-effects I'm not aware of, that require this to match the mapping exactly

Reply:
> There is no technological reason why TS types must === mapping.
Of course. This is just to know what to expect inside the indexed docs. I wasn't aware that the endpoint is writing this to the index.

> Unmapped fields are still stored, are still JSON, will still be sent to the frontend who can parse the JSON and therefore know about the property.
Right again.  

> I don't know the rest of the context here, whether this type is being used in a filter somewhere that will otherwise inform part of the UI what filters are available (unmapped means not filterable), or if there are similar side-effects I'm not aware of, that require this to match the mapping exactly